qutebrowser/keyinput/basekeyparser.py

```diff
@@ -104,33 +191,36 @@ def _match_key(self, sequence):
         """
         assert sequence
         assert not isinstance(sequence, str)
-        result = QKeySequence.NoMatch
-
-        for seq, cmd in self.bindings.items():
-            assert not isinstance(seq, str), seq
-            match = sequence.matches(seq)
-            if match == QKeySequence.ExactMatch:
-                return match, cmd
-            elif match == QKeySequence.PartialMatch:
-                result = QKeySequence.PartialMatch
-
-        return result, None
-
-    def _match_without_modifiers(self, sequence):
+        override_result = (QKeySequence.NoMatch, None)
+        overrides = config.instance.get(
+            'bindings.commands', url=url, fallback=False)
```

Comment:
Just a heads up - this line is extremely costly (~40% of time spent when typing, to the point of causing lag when typing), but with #4707 this is only bad when there are per-domain bindings that could be matched. I might work on config cache for per-domain bindings just for this line.